src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs

```diff
@@ -60,7 +59,16 @@ public async Task ServerAsyncAuthenticate_EachSupportedProtocol_Success(SslProto
                 });
 
             Assert.NotNull(e);
-            Assert.IsType<AuthenticationException>(e);
+
+            if (OperatingSystem.IsAndroid() && (PlatformDetection.IsX64Process || PlatformDetection.IsX86Process))
+            {
+                // On Android running on x64 or x86 the server side sometimes throws IOException instead of AuthenticationException
+                Assert.True(typeof(IOException) == e.GetType() || typeof(AuthenticationException) == e.GetType(), $"Unexpected exception type: {e.GetType()}");
```

Comment:
`e is IOException || e is AuthenticationException` ?
Also this seems just like timing, right? I'm wondering if you really need the architecture check - I would keep it simple.